client/tools/mgr-push/test/test_uploadclass.py

```diff
@@ -0,0 +1,315 @@
+    )
+    @patch("rhnpush.rhnpush_cache.RHNPushSession", Mock())
+    @patch("up2date_client.rhnserver.RhnServer", Mock())
+    @patch("rhnpush.uploadLib.call", Mock(side_effect=["123", 0]))
+    def test_packages(self):
+        server_digest_hash = {
+            "existentPackage_1.0-1_amd64.deb": [
+                "sha256",
+                "cd5b4348e7b76c2287a9476f3f3ef3911480fe8e872ac541ffb598186a9e9607",
+            ],
+            "newPackage_2.0-1_amd64.deb": "",
+        }
+
+        pkgs_info = {
+            "existentPackage_1.0-1_amd64.deb": {
+                "name": "existentPackage",
+                "version": "1.0",
+                "release": "1",
+                "epoch": "",
+                "arch": "amd64-deb",
+                "checksum_type": "sha256",
+                "checksum": "cd5b4348e7b76c2287a9476f3f3ef3911480fe8e872ac541ffb598186a9e9607",
+            },
+            "newPackage_2.0-1_amd64.deb": {
+                "name": "newPackage",
+                "version": "2.0",
+                "release": "1",
+                "epoch": "",
+                "arch": "amd64-deb",
+                "checksum_type": "sha256",
+                "checksum": "ece4eedf7a5c65396d136b5765226e2c8b10f268c744b0ab1fa2625e35384a00",
+            },
+        }
+
+        digest_hash = {
+            "existentPackage_1.0-1_amd64.deb": (
+                "sha256",
+                "cd5b4348e7b76c2287a9476f3f3ef3911480fe8e872ac541ffb598186a9e9607",
+            ),
+            "newPackage_2.0-1_amd64.deb": (
+                "sha256",
+                "ece4eedf7a5c65396d136b5765226e2c8b10f268c744b0ab1fa2625e35384a00",
+            ),
+        }
+
+        with patch(
+            "rhnpush.rhnpush_main.UploadClass.package", Mock(return_value=0)
+        ) as package, patch(
+            "rhnpush.rhnpush_main.UploadClass.check_package_exists",
+            Mock(return_value=(server_digest_hash, pkgs_info, digest_hash)),
+        ), patch(
+            "rhnpush.uploadLib.listdir",
+            Mock(return_value=["newPackage_2.0-1_amd64.deb"]),
+        ):
+            config_parser = rhnpush_config.rhnpushConfigParser()
+            with (patch.dict(config_parser.__dict__, TestUpload.config)):
+                self._upload.options = config_parser
+                self._upload.directory()
+
+                self._upload.packages()
+                package.assert_called_with(
+                    "newPackage_2.0-1_amd64.deb",
+                    "sha256",
+                    "ece4eedf7a5c65396d136b5765226e2c8b10f268c744b0ab1fa2625e35384a00",
+                )
+
+                # test call to uploadLib.call(self.server.packages.channelPackageSubscriptionBySession, self.session.getSessionString(), info)
+                # info [{'name': 'avra', 'version': '1.3.0', 'release': '3', 'epoch': '', 'arch': 'amd64-deb', 'checksum_type': 'sha256', 'checksum': 'e08f76afa7bd472a3a7db84874a208bda3ca59b4f6784136a31df35dcf231ec6'}]
```

Comment:
Is something missing here?

Reply:
That's a leftover that I wanted to remove, thank you for noticing.